src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -0,0 +1,723 @@
+/****************************************************************************************/
+
+
+/// TODO: Edit for NEURON
+void CodegenNeuronCppVisitor::print_nrn_current(const BreakpointBlock& node) {
+    return;
+}
+
+
+/// TODO: Edit for NEURON
+void CodegenNeuronCppVisitor::print_nrn_cur_conductance_kernel(const BreakpointBlock& node) {
+    return;
+}
+
+
+/// TODO: Edit for NEURON
+void CodegenNeuronCppVisitor::print_nrn_cur_non_conductance_kernel() {
+    return;
+}
+
+
+/// TODO: Edit for NEURON
+void CodegenNeuronCppVisitor::print_nrn_cur_kernel(const BreakpointBlock& node) {
+    return;
+}
+
+
+/// TODO: Edit for NEURON
+void CodegenNeuronCppVisitor::print_fast_imem_calculation() {
+    return;
+}
+
+
+/// TODO: Edit for NEURON
+void CodegenNeuronCppVisitor::print_nrn_cur() {
+    if (!nrn_cur_required()) {
+        return;
+    }
+
+    codegen = true;
+
+    printer->add_line("void nrn_cur() {}");
+    /// TODO: Fill in
+
+    codegen = false;
+}
+
+
+/****************************************************************************************/
+/*                            Main code printing entry points                            */
+/****************************************************************************************/
+
+void CodegenNeuronCppVisitor::print_headers_include() {
+    print_standard_includes();
+    print_neuron_includes();
+}
+
+
+void CodegenNeuronCppVisitor::print_macro_definitions() {
+    print_global_macros();
+    print_mechanism_variables_macros();
+}
+
+
+void CodegenNeuronCppVisitor::print_global_macros() {
+    printer->add_newline();
+    printer->add_line("/* NEURON global macro definitions */");
+    if (info.vectorize) {
+        printer->add_multi_line(R"CODE(
+            /* VECTORIZED */
+            #define NRN_VECTORIZED 1
+        )CODE");
+    } else {
+        printer->add_multi_line(R"CODE(
+            /* NOT VECTORIZED */
+            #define NRN_VECTORIZED 0
+        )CODE");
+    }
+}
+
+
+void CodegenNeuronCppVisitor::print_mechanism_variables_macros() {
+    printer->add_newline();
+    printer->add_line("static constexpr auto number_of_datum_variables = ",
+                      std::to_string(int_variables_size()),
+                      ";");
+    printer->add_line("static constexpr auto number_of_floating_point_variables = ",
+                      std::to_string(float_variables_size()),
+                      ";");
+    printer->add_newline();
+    printer->add_multi_line(R"CODE(
+    namespace {
+    template <typename T>
+    using _nrn_mechanism_std_vector = std::vector<T>;
+    using _nrn_model_sorted_token = neuron::model_sorted_token;
+    using _nrn_mechanism_cache_range =
+        neuron::cache::MechanismRange<number_of_floating_point_variables, number_of_datum_variables>;
+    using _nrn_mechanism_cache_instance =
+        neuron::cache::MechanismInstance<number_of_floating_point_variables, number_of_datum_variables>;
```

Comment:
`another nitpick/suggestion`: while printing code here, maybe we can print without enforcing ~100 column width limit i.e. print these on single line so that it's easier to read this code here.